repository/repository.go

```diff
@@ -0,0 +1,19 @@
+package repository
+
+import (
+	"github.com/murilogilfelpeto/ps-tag-onboarding-go/configuration"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+var (
+	logger         *configuration.Logger
+	userCollection *mongo.Collection
+	context        = configuration.GetContext()
+)
+
```

Comment:
While it's not inherently wrong to use global variables, and there are scenarios where they might be appropriate, the overuse of global variables can lead to many issues.

* Modular code is often preferred. For example, the use of a `New()` constructor where you can inject the dependencies
* Code becomes less readable when variables are not localized to the scope where they are used.

Some tips:
1. Try to use more function parameters and return values to pass data between functions
2. [Organize code into well defined modules with clear interfaces](https://github.com/wexinc/ps-tag-ledger/blob/main/internal/common/runner/runner.go#L37)
3. [Dependency injection is always your friend](https://github.com/wexinc/ps-tag-ledger/blob/main/internal/common/runner/runner.go#L55)
4. [The use of configuration objects](https://github.com/wexinc/ps-tag-payment-scheduler/blob/main/internal/adapters/mongodb/mongo.go#L17)

See some examples here:
https://github.com/wexinc/ps-tag-ledger/blob/main/internal/domain/financial/repo/financial.go#L26
https://github.com/wexinc/ps-tag-non-posted-exceptions/blob/main/internal/repo/repo.go